### ISSUE TITLE
Some problems in training

### ISSUE THREAD

**Issue by zhaoyun-ai** [edufonseca/uclser#2]
I run the code on Ubuntu20.04，but i get Loss: 0.0000, Accuracy: nan when I run python main_train.py -p config/params_unsupervised_cl.yaml. I get  Accuracy: 100% when I run python main_train.py -p config/params_supervised_lineval.yaml.
I don't know the reason for the mistake.

python main_train.py -p config/params_unsupervised_cl.yaml：
############# Data loaded #############
Total params: 11.18 M
######## Validation ########

Validation set. Loss: 0.0000, Accuracy: nan

######## Test ########

Test set. Loss: 0.0000, Accuracy: nan

Epoch time: 142.33 seconds

######## Validation ########

Validation set. Loss: 0.0000, Accuracy: nan

python main_train.py -p config/params_supervised_lineval.yaml：
############# Data loaded #############
Total params: 11.18 M
######## Epoch: 1
Train Epoch: 1 [1920/15813 (12%)]       Loss: 0.000446, Accuracy: 100%, Learning rate: 0.100000
Train Epoch: 1 [3840/15813 (24%)]       Loss: 0.000027, Accuracy: 100%, Learning rate: 0.100000
Train Epoch: 1 [5760/15813 (36%)]       Loss: 0.000008, Accuracy: 100%, Learning rate: 0.100000
Train Epoch: 1 [7680/15813 (48%)]       Loss: 0.000009, Accuracy: 100%, Learning rate: 0.100000
Train Epoch: 1 [9600/15813 (60%)]       Loss: 0.000007, Accuracy: 100%, Learning rate: 0.100000
Train Epoch: 1 [11520/15813 (73%)]      Loss: 0.000009, Accuracy: 100%, Learning rate: 0.100000
Train Epoch: 1 [13440/15813 (85%)]      Loss: 0.000006, Accuracy: 100%, Learning rate: 0.100000
Train Epoch: 1 [15360/15813 (97%)]      Loss: 0.000010, Accuracy: 100%, Learning rate: 0.100000
######## Validation ########

Validation set. Loss: 0.0000, Accuracy: nan

######## Test ########

Test set. Loss: 0.0000, Accuracy: nan

Epoch time: 56.20 seconds

**Comment by Adhithyan-Ramadoss** [comment]
> 
@[zhaoyun-ai](https://github.com/zhaoyun-ai)
did you modify the number of classes (num_classes) in the configuration file. I noticed that when your test/val set is operating on incorrect number of classes, you get a division by 0 leading to a nan in the accuracy calculation (see src/utils_train_eval, lines 323-325).
